prometheus_client/metrics.py

```diff
@@ -73,6 +73,14 @@ def collect(self):
             metric.add_sample(self._name + suffix, labels, value)
         return [metric]
 
+    def reset(self):
+        if self._is_parent():
+            with self._lock:
+                for labels, metric in self._metrics.items():
+                    metric.reset()
+        elif self._is_observable():
+            self._metric_init()
```

Comment:
It took me a while to parse this so I want to share . . . 

Instances of `MetricWrapperBase` or metrics like `Histogram` that inherits from it corresponds to a metric name like "lambda_execution_time".  The instance includes a dictionary called `_metrics`.  This dictionary holds the values associated with each label/value pair.  You can see in the `labels` method where we look a metric in `metrics` based on the label values.

The actual value of a metric is stored on the `Metric` instance in that `_metrics` dictionary and is itself an instance of `MutexValue` which wraps getting and setting the metric in a lock.

The exception to this is if a metric has no labels, in that case it won't have `_metrics` dictionary.

For example:
```
>>> import prometheus_client as pc
>>> c = pc.Counter('example', 'Example', ['animal'])
>>> c.labels('horse').inc()
>>> c._metrics
{('horse',): <prometheus_client.metrics.Counter object at 0x10782dc18>}
>>> c.labels('horse')
<prometheus_client.metrics.Counter object at 0x10782dc18>
>>> c.labels('horse')._value
<prometheus_client.values.MutexValue object at 0x10782dc50>
>>> c.labels('horse')._value._value
1.0
```